File: 1700-minimum-time-to-make-rope-colorful/1700-minimum-time-to-make-rope-colorful.java
class Solution {
    public int minCost(String colors, int[] neededTime) {
        int total = 0;
        int maxInGroup = neededTime[0];

        for (int i = 1; i < colors.length(); i++) {
            if (colors.charAt(i) == colors.charAt(i - 1)) {
                total += Math.min(neededTime[i], maxInGroup);
                maxInGroup = Math.max(maxInGroup, neededTime[i]);
            } else {
                maxInGroup = neededTime[i];
            }
        }

        return total;
    }
}

File: 1700-minimum-time-to-make-rope-colorful/1700-minimum-time-to-make-rope-colorful.rs
impl Solution {
    pub fn min_cost(colors: String, needed_time: Vec<i32>) -> i32 {
        let mut total=0;
        let bytes = colors.as_bytes();
        let mut max_group = needed_time[0];

         for i in 1..bytes.len() {
            if bytes[i] == bytes[i - 1] {
                total += needed_time[i].min(max_group);
                max_group = max_group.max(needed_time[i]);
            } else {
                max_group = needed_time[i];
            }
        }

        total
    }
}